docs/ossm/ossm2-migration/gateway-migration.md

```diff
@@ -0,0 +1,92 @@
+# Gateway Migration Guide
+
+Migrating gateways between Istio control planes during a version upgrade from 2.6 to 3.0 is very similar to migrating regular workloads, here is some information on how to migrate your gateways.
+
+## Migration Scenarios
+
+### Gateway Canary Migration (Recommended)
```

Comment:
Do we need to recommend the canary migration? It's considered "advanced" in the upstream docs: https://istio.io/latest/docs/setup/additional-setup/gateway/#canary-upgrade-advanced and doing an in place upgrade shouldn't cause any downtime.

Reply:
I'm not sure about the downtime. I have a test which is sending 10 req/sec through the gateway when the gateway is being restarted during the migration and sometimes there is one 502 response code. But I don't why it's happening yet so the expectation there should be no downtime even for inplace migration probably still applies.

Reply:
I listed it as recommended since its what we [recommended](https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/service_mesh/service-mesh-2-x#ossm-migrating-from-smcp-defined-gateways-to-gateway-injection_gateway-migration) when switching from SMCP controlled gws to gw injection gws and filip suggested it was safest option for bigger customers using shared gateways.

Reply:
Both Advanced or Recommended are fine with me.